rest/rest.go

```diff
@@ -51,6 +51,7 @@ type EndpointType client.EndpointType
 
 // Server contains configuration and handlers for additional listeners to be instantiated after app startup.
 type Server struct {
+	CoreAPI     bool
```

Comment:
Thought about checking if the `Address` is specified to determine if a `Server` should be grouped with the core API, but that complicates things since the address of a network listener should be set in the `PreBootStrap` or `PreJoin` hooks. Having the `CoreAPI` flag makes the intention explicit that the server and its endpoints should be added to the core API network listener.